### DISCUSSION TITLE
Curved or Arched Text

### DISCUSSION THREAD

**Discussion by lcundiff** [hm21/pro_image_editor#251] [Ideas]
Amazing library! Love the customizations and enhanced text editing. I'm not sure how difficult it would be, but enabling the arching/curving of text would allow for the ability to make images like these. 

![image](https://github.com/user-attachments/assets/b58701ff-217a-4a54-b43c-e6d505feb93b)

**Comment by hm21** [comment]
Currently, it’s not possible to curve text directly. However, you can create a custom solution using a package like [flutter_circular_text](https://pub.dev/packages/flutter_circular_text), allowing you to build your own sub-editor. Once that’s set up, you can add the widget as a layer to the editor with code like `editorKey.currentState!.addLayer(CurvedText('Text'));`.

Let me know if you’d like a more detailed explanation. I can provide a step-by-step guide to help you implement it in the case you need it.

**Reply by lcundiff** [reply]
Okay thank you, that makes sense. If I have anymore follow up questions, I'll add them here :)
